custom-graders/DemoAssignmentGrader/GraderFiles/executeGrader.sh

```diff
@@ -0,0 +1,75 @@
+#! /bin/bash
+
+# Coursera deletes all environment variables set inside 'Dockerfile'. If any environment varables
+# need to be set, they must be set inside a wrapper bash script.
+export JAVA_HOME=/usr/lib/jvm/java-7-openjdk-amd64
+
+# Switch to the grader directory
+cd /grader
+
+# Compile the learner's program in the current directory. We can safely assume that there
+# would be a single submission file in this directory.
+javac -d . /shared/submission/*
+
+# Note: Nothing except Json object containing 'fractionalScore' and 'feedback' should be written
+# to stdout.
+
+# Check if the compilation was successful
+if [ ! $? -eq 0 ]; then
+    echo "{ \"fractionalScore\":0.0, \"feedback\":\"Compile Error\" }"
+    exit 0
+fi
+
+# Parse the command line arguments supplied by Coursera.
+while [[ $# > 1 ]]
+do
+key="$1"
+
+case $key in
+  partId)
+  # Unique Id associated with the part which is being graded.
+  PARTID="$2"
+  shift
+  ;;
+  userId)
+  # Unique integer Id of the learner that made this submission.
+  USERID="$2"
+  shift
+  ;;
+  filename)
+  # Original filename as uploaded by the learner before it was renamed to the suggested filename.
+  ORIGINAL_FILENAME="$2"
+  shift
+  ;;
+esac
+shift
+done
+
+# Use the parsed partId to know which part is being graded in the current run.
+if [ "$PARTID" == "HxbKF" ]; then
```

Comment:
Please comment where the constants come from. (Ideally, pull them out into named variables defined at the top of the file.)